core/integration/remotecache_test.go

```diff
@@ -293,7 +304,7 @@ func runSeparateEngine(ctx context.Context, t *testing.T, env map[string]string,
 	require.NoError(t, err)
 
 	return c, func() error {
-		out, err := exec.Command("docker", "stop", "-s", "SIGTERM", "-t", "30", name).CombinedOutput()
+		out, err := exec.Command("docker", "kill", "-s", "SIGTERM", name).CombinedOutput()
```

Comment:
Saw this was for compatibility, which makes sense, but I'm wondering if this is also the cause of the tests seemingly getting blocked forever... If it is that indicates something is wrong with engine shutdown, which I'm happy to look into since there's a good chance it's related to my changes that require synchronizing remote cache stuff on shutdown.

I'll go do that but if you want to unblock immediately you could try putting a timeout on `docker wait` below and then doing a SIGKILL if it didn't exit in time.

Reply:
Update: pushed this commit with a fix, looks like engine tests pass now, https://github.com/dagger/dagger/pull/4711/commits/954b38002625bfab93633881e378eb085e69df17

The problem was just that in those test cases I wrote previously I forgot to close the dagger clients, which ended up blocking the engine on `GracefulShutdown`: https://github.com/dagger/dagger/blob/d275122c01bcb585b54aa212ebaee9518d4eb438/cmd/engine/main.go#L354